lib/kredis/types/counter.rb

```diff
@@ -24,4 +20,17 @@ def value
   def reset
     del
   end
+
+  private
+
+    def value=(new_value)
+      set(new_value.to_i, ex: expires_in, nx: true)
+    end
+
+    def set_default
+      return if exists?
+
+      value = @default.is_a?(Proc) ? @default.call : @default
+      self.value = value
```

Comment:
Why the local `value` variable? Can just assign directly. You don't need the guard clause either:

```ruby
self.value = resolve_default_value unless exists?
```

And then add `resolve_default_value`.